File: Number-System/3079-find-the-sum-of-encrypted-integers.py
# Time  : O(n)
# Space : O(1)

class Solution:
  def sumOfEncryptedInt(self, A: List[int]) -> int:
    sum = 0
    
    def encrypt(n):
      ma = 1
      digits = 0
      while n:
        digits += 1
        digit = n % 10
        ma = max(ma, digit)
        n //= 10
      ret = 0
      for i in range(digits):
        ret = ret * 10 + ma
      return ret
    
    for n in A:
      sum += encrypt(n)
      
    return sum

File: Number-System/3079-find-the-sum-of-encrypted-integers.cpp
// Time  : O(n)
// Space : O(1)
class Solution {
public:
  int sumOfEncryptedInt(vector<int>& A) {
    int ans = 0;
    for (int& a : A) {
      int digits = 0;
      int maxdigit = 1;
      while (a) {
        int digit = a % 10;
        ++digits;
        maxdigit = max(maxdigit, digit);
        a /= 10;
      }
      int aa = 0;
      while (digits--) {
        aa = aa * 10 + maxdigit;
      }
      ans += aa;
    }
    return ans;
  }
};